### PR TITLE
test(yaml): add test for YAML tags

### PR DESCRIPTION
**Pull request by websi**
## Changes

Add generic support for YAML tags.

## Context

See https://github.com/renovatebot/renovate/discussions/31147

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by websi** [comment]
@viceice is this needed anymore if #31336 is merged.
Maybe we should  also at an testcase for docker-compose yaml tags within #31336.

**Comment by websi** [comment]
@rarkins @viceice #31336 seems not to fix this problem.

What is the next step? Should i rebase and adapt the changes?

**Comment by viceice** [comment]
> @rarkins @viceice #31336 seems not to fix this problem.
> 
> What is the next step? Should i rebase and adapt the changes?

you need to adapt the code for the new yaml parser. it should now be possible to support custom yaml tags without removing them.

**Review by viceice — CHANGES_REQUESTED** [review]
see other comment

**Comment by websi** [comment]
I tested the current master against https://github.com/websi/renovate-reproductions-31147. Everything works with the new library. I add some new test cases for the YAML tags and remove the extra handling for the GitLab Reference tag

File: lib/modules/manager/docker-compose/extract.spec.ts [hunk 1]
```diff
@@ -56,6 +57,23 @@ describe('modules/manager/docker-compose/extract', () => {
       expect(res?.deps).toHaveLength(1);
     });
 
+    it('extracts can parse yaml tags for version 3', () => {
+      const res = extractPackageFile(yamlFile3WithTags, '', {});
+      expect(res?.deps).toMatchInlineSnapshot(`
```
Comment (viceice) [inline lib/modules/manager/docker-compose/extract.spec.ts:62]: use `toEqual` or `toMatchObject`

File: lib/modules/manager/docker-compose/__fixtures__/docker-compose.3-tags.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+services:
```
Comment (viceice) [inline lib/modules/manager/docker-compose/__fixtures__/docker-compose.3-tags.yml:1]: reduce to minimal (one service) and inline to test

File: lib/modules/manager/gitlabci-include/extract.ts [hunk 3]
```diff
@@ -72,7 +71,7 @@ export function extractPackageFile(
   const endpoint = GlobalConfig.get('endpoint');
   try {
     // TODO: use schema (#9610)
-    const doc = parseSingleYaml<GitlabPipeline>(replaceReferenceTags(content));
+    const doc = parseSingleYaml<GitlabPipeline>(content);
```
Comment (viceice) [inline lib/modules/manager/gitlabci-include/extract.ts:74]: move to seperate PR, this is a refactor.

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/modules/manager/docker-compose/__fixtures__/docker-compose.3-tags.yml (hunk 2 above)
Comment (websi) [reply lib/modules/manager/docker-compose/__fixtures__/docker-compose.3-tags.yml:1]: if i reduce the test case to one service, the docker compose file ist not a valid compose file anymore.
Should this ok?

You need minimum two services to write an valid extend and override configuration.

File: lib/modules/manager/gitlabci-include/extract.ts (hunk 3 above)
Comment (websi) [reply lib/modules/manager/gitlabci-include/extract.ts]: See https://github.com/renovatebot/renovate/pull/31787

**Review by viceice — APPROVED** [review]

**Pull request merged**